juju/application.py

```diff
@@ -217,16 +217,31 @@ async def destroy_unit(self, *unit_names):
         return await self.model.destroy_units(*unit_names)
     destroy_units = destroy_unit
 
-    async def destroy(self):
+    async def destroy(self, destroy_storage=False, force=False, no_wait=False):
         """Remove this application from the model.
 
+        :param bool destroy_storage: Destroy storage attached to application unit. (=false)
+        :param bool force: Completely remove an application and all its dependencies. (=false)
+        :param bool no_wait: Rush through application removal without waiting for each individual step to complete (=false)
+        :param bool block: Blocks until the application is removed from the model
         """
+
+        if no_wait and not force:
```

Comment:
Following the behaviour from the CLI, `no-wait` and `force` should not be set by default.